tests/test_evaluator.py

```diff
@@ -1 +1,175 @@
 """Unit tests for the `evaluator` module."""
+
+from collections import defaultdict
+from typing import Dict
+
+import pytest
+
+from aiai_eval.utils import Device
+from src.aiai_eval.evaluator import Evaluator
+from src.aiai_eval.exceptions import (
+    InvalidArchitectureForTask,
+    ModelDoesNotExistOnHuggingFaceHub,
+)
+from src.aiai_eval.task_configs import SENT
+from src.aiai_eval.task_factory import TaskFactory
+
+
+@pytest.fixture(scope="module")
+def evaluator():
+    evaluator = Evaluator(
+        progress_bar=True,
+        save_results=False,
+        raise_error_on_invalid_model=False,
+        cache_dir=".aiai_cache",
+        use_auth_token=False,
+        track_carbon_emissions=False,
+        country_iso_code="",
+        prefer_device=Device.CPU,
+        verbose=False,
+    )
+    evaluator.evaluation_config.testing = True
+    yield evaluator
+
+
+@pytest.fixture(scope="module")
+def non_existing_model_id():
+    yield "invalid-model-id"
+
+
+@pytest.fixture(scope="module")
+def existing_model_id():
+    yield "bert-base-uncased"
+
+
+class TestPrepareModelIds:
+    def test_prepare_model_ids(self, evaluator, existing_model_id):
+        model_ids = evaluator._prepare_model_ids([existing_model_id, existing_model_id])
+        assert model_ids == [existing_model_id, existing_model_id]
+        model_ids = evaluator._prepare_model_ids(existing_model_id)
+        assert model_ids == [existing_model_id]
+
+
+class TestPrepareTaskConfig:
+    def test_prepare_task_config_list_task(self, evaluator):
+        task_config = evaluator._prepare_task_configs(["sent", "sent"])
+        assert task_config == [SENT, SENT]
+
+    def test_prepare_task_config_str_task(self, evaluator):
+        task_config = evaluator._prepare_task_configs("sent")
+        assert task_config == [SENT]
```

Comment:
Using the global `task_config` fixture:

```suggestion
class TestPrepareTaskConfig:
    def test_prepare_task_config_list_task(self, evaluator, task_config):
        task_config = evaluator._prepare_task_configs([task_config.name, task_config.name])
        assert task_config == [task_config, task_config]

    def test_prepare_task_config_str_task(self, evaluator, task_config):
        task_config = evaluator._prepare_task_configs(task_config.name)
        assert task_config == [task_config]
```